File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int ans = 0, left=0, right=height.length-1;
        while(left<right){
            if(height[left]<height[right]){
                ans = Math.max(ans, (right-left)*Math.min(height[left],height[right]));
                left++;
            }
            else { ans = Math.max(ans, (right-left)*Math.min(height[left],height[right]));
                right--;}
        }
        return ans;
    }
        
}

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area=0
        left=0
        right=len(height)-1
        while(left<right):
            if height[left]<height[right]:
                area=max(area, height[left]*(right-left))
                left+=1
            else:
                area=max(area, height[right]*(right-left))
                right-=1
        return area